alerts/HighlyAvailableWorkloadIncorrectlySpread.md

```diff
@@ -0,0 +1,89 @@
+# HighlyAvailableWorkloadIncorrectlySpread
+
+## Meaning
+
+Highly available workloads with persistent storage are incorrectly spread across
+multiple nodes. In high availability topology, all workloads with multiple
+replicas should be spread across multiple nodes to avoid having a single point
+of failure. However, workload with persistent storage are more complex to
+schedule since persistent volumes may be bound to an availability zone, so they
+need to be spread manually before any scheduling constraints are put in place.
+
+## Impact
+
+The cluster isn't fully highly-available since workloads with persistent
+storage might still have a single point of failure.
+
+## Diagnosis
+
+The alert indicates which workloads aren't correctly spread across multiple
+nodes. You can find information to identify them under the `workload` and
+`namespace` labels.
+
+```console
+ - alertname = HighlyAvailableWorkloadIncorrectlySpread
+...
+ - workload = prometheus-k8s
+...
+ - namespace = openshift-monitoring
+```
+
+Then, you can verify that multiple instances of the same application are indeed
+scheduled on the same node by running:
+
+```console
+$ oc -n "$NS" get -o wide pods | grep "$WORKLOAD"
+```
+
+## Mitigation
+
+The script below is provided to rebalance workloads across multiple nodes. To
+use it, you should first get the following information from the alert labels:
+* `workload`: workload name
+* `namespace`: namespace of the workload
+* `node`: node hosting multiple instances of the workload
+
+Then you will want to update the script to include the workloads that need to be
+rebalance across multiple nodes. To do so you can add the following call for
+each workload reported by the alert at the end of the script below:
+
+```sh
+rebalancePods $WORKLOAD $NS $NODE
+```
+
+For example if you need to rebalance `prometheus-k8s`, you would have to add the
+following line at the end of the script:
+```sh
+rebalancePods prometheus-k8s openshift-monitoring ip-10-0-0-0.ec2.internal
+```
+
+```sh
+#!/bin/bash
+
+AZ_COUNT=$(oc get nodes -o yaml | grep -E "^\s+failure-domain.beta.kubernetes.io/zone" | uniq | wc -l)
+
+rebalancePods() {
+    WORKLOAD=$1
+    NS=$2
+    NODE=$3
+
+    echo "Cordon node $NODE"
+    oc adm cordon "$NODE"
```

Comment:
This shouldn't be necessary since we have soft anti-affinity, but I think that this is a good safety measure to avoid having pods scheduled on the same node again.